L1Trigger/TrackFindingTMTT/src/L1KalmanComb.cc

```diff
@@ -0,0 +1,1829 @@
+            if (hphiErrorRatioMap.find(hname) == hphiErrorRatioMap.end()) {
+              cout << hname << " does not exist." << endl;
+            } else {
+              hphiErrorRatioMap[hname]->Fill(fabs(stbcl->deltai() + 0.5), fabs(stbcl->dphi_dr()) / stbcl->dphi_dl());
+            }
+          }
+        }
+      }
+    }
+    if (getSettings()->kalmanFillInternalHists()) {
+      if (tpa && tpa->useForAlgEff()) {
+        hTrackEta_->Fill(tpa->eta());
+        static thread_local set<const TP *> set_tp;
+        if (iCurrentPhiSec_ < iLastPhiSec_ && iCurrentEtaReg_ < iLastEtaReg_)
+          set_tp.clear();
+        if (set_tp.find(tpa) == set_tp.end()) {
+          hUniqueTrackEta_->Fill(tpa->eta());
+        }
+        set_tp.insert(tpa);
+      }
+    }
+
+    //track information dump
+    if (getSettings()->kalmanDebugLevel() >= 1) {
+      std::cout << "===============================================================================" << endl;
+      std::cout << "Input track cand: [phiSec,etaReg]=[" << l1track3D.iPhiSec() << "," << l1track3D.iEtaReg() << "]";
+      std::cout << " HT(m,c)=(" << l1track3D.getCellLocationHT().first << "," << l1track3D.getCellLocationHT().second
+                << ") q/pt=" << l1track3D.qOverPt() << " tanL=" << l1track3D.tanLambda() << " z0=" << l1track3D.z0()
+                << " phi0=" << l1track3D.phi0() << " nStubs=" << l1track3D.getNumStubs() << " d0=" << l1track3D.d0()
+                << std::endl;
+      if (not getSettings()->hybrid())
+        printTP(cout, tpa);
+      if (getSettings()->kalmanDebugLevel() >= 2) {
+        printStubLayers(cout, stubs, l1track3D.iEtaReg());
+        printStubClusters(cout, stubcls);
+      }
+    }
+
+    //Kalman Filter
+    std::vector<const KalmanState *> cands = doKF(l1track3D, stubcls, tpa);
+
+    //return L1fittedTrk for the selected state (if KF produced one it was happy with).
+    if (cands.size()) {
+      const KalmanState *cand = cands[0];
+
+      //cout<<"Final KF candidate eta="<<cand->candidate().iEtaReg()<<" ns="<<cand->nSkippedLayers()<<" klid="<<cand->nextLayer()-1<<" n="<<cand->nStubLayers()<<endl;
+
+      // Get track helix params.
+      std::map<std::string, double> trackParams = getTrackParams(cand);
+
+      L1fittedTrack returnTrk(getSettings(),
+                              l1track3D,
+                              cand->stubs(),
+                              cand->hitPattern(),
+                              trackParams["qOverPt"],
+                              trackParams["d0"],
+                              trackParams["phi0"],
+                              trackParams["z0"],
+                              trackParams["t"],
+                              cand->chi2rphi(),
+                              cand->chi2rz(),
+                              nPar_,
+                              true);
+
+      bool consistentHLS = false;  // No longer used
+      //    if (this->isHLS()) {
```

Comment:
An HLS version of the KF also exists, that we have run privately within CMSSW. To simplify the current PR, we have deleted the HLS version of the KF. It will be added to a future CMSSW release as an external library. The commented out code, such as this, which mentions HLS are the interface to this external library. If we delete them all, it will be a lot of work to put them back in future. Can we keep them?

Reply:
Any commented-out lines that should be kept should be marked with a comment explaining why

Reply:
done